File: src/main/java/ninja/hudy/infosmog/repository/StationRepository.java
package ninja.hudy.infosmog.repository;

import ninja.hudy.infosmog.model.Station;
import ninja.hudy.infosmog.type.StationType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface StationRepository extends JpaRepository<Station, Long> {

    List<Station> findByType(StationType type);
    Optional<Station> findByTypeAndProviderCode(StationType type, String providerCode);
}

File: src/main/kotlin/pl/org/smok/infosmog/repository/StationRepository.kt
package pl.org.smok.infosmog.repository

import pl.org.smok.infosmog.model.Station
import pl.org.smok.infosmog.type.StationType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface StationRepository : JpaRepository<Station, Long> {
    fun findByType(type: StationType): List<Station>
    fun findByTypeAndProviderCode(type: StationType, providerCode: String): Optional<Station>
}
